app/models/submission.rb

```diff
@@ -400,25 +400,9 @@ def as_json(options = {})
   end
 
   def grading_complete?(as_seen_by)
-    include_unreleased = !as_seen_by.student?
-
-    complete, released = scores_status
-    (released || include_unreleased) && complete
-  end
-
-  def scores_status
-    all_complete = true
-    all_released = true
-
-    problems_to_scores.each do |problem, score|
-      next if problem.optional?
-      return false unless score
-
-      all_complete &&= false unless score.score
-      all_released &&= score.released?
-    end
-
-    [all_complete, all_released]
+    include_unreleased = as_seen_by.course_assistant? || as_seen_by.instructor?
+    released = scores.pluck(:released).all?
+    released || include_unreleased
```

Comment:
Previously, there was an additional check that all scores were non-nil. But this would mean that grading would not be deemed complete unless instructors grade every single manual problem.

Given that grading_complete? is not currently used, I have adapted it to suit the purposes of the rest of the code.